scalafix/output/src/main/scala/rsc/tests/BetterRscCompat.scala

```diff
@@ -17,6 +17,10 @@ object BetterRscCompat_Test {
     def map: mutable.Map[Int, String] = mutable.Map(1 -> "1")
     def str: String = "hello"
     def str2: String = s"$str world"
+    val foo: Int => Int => Int = (x: Int) => (y: Int) => x + y
+    val bar: (Int => Int) => Int => Int = (f: Int => Int) => f
+    val baz: Int => () => (Int => Int) => Int => Int = (f: Int) => bar _
```

Comment:
We can try an example with higher arity, `def`, and polymorphism: `def foo[A]: (A, A => A) => A = (x: A, y: A => Int) => x`